CMakeLists.txt

```diff
@@ -393,6 +393,21 @@ include_directories(SYSTEM ${LIBLUV_INCLUDE_DIRS})
 find_package(TreeSitter REQUIRED)
 include_directories(SYSTEM ${TreeSitter_INCLUDE_DIRS})
 
+list(APPEND CMAKE_REQUIRED_INCLUDES "${TreeSitter_INCLUDE_DIRS}")
+list(APPEND CMAKE_REQUIRED_LIBRARIES "${TreeSitter_LIBRARIES}")
+check_c_source_compiles("
+#include <tree_sitter/api.h>
+int
+main(void)
+{
+  TSQueryCursor *cursor = ts_query_cursor_new();
+  ts_query_cursor_set_match_limit(cursor, 32);
```

Comment:
maybe add `return 0` in case of overzealous non-void missing return error (unlikely but better safe than sorry)